core/org.wso2.carbon.registry.core/src/main/java/org/wso2/carbon/registry/core/jdbc/dataaccess/JDBCTransactionManager.java

```diff
@@ -94,6 +94,14 @@ public void beginTransaction() throws RegistryException {
         } catch (SQLException e) {
             String msg = "Failed to start new registry transaction.";
             log.error(msg, e);
+            if (conn != null) {
+                try {
+                    conn.close();
+                } catch (SQLException e1) {
+                    String msgFailed = "Failed to close connection.";
+                    log.error(msgFailed, e);
+                }
+            }
```

Comment:
I think I misunderstood the logic, as the purpose of this method is to start a transaction, closing the connection using a finally block would be wrong.